pkg/kubelet/cri/.import-restrictions

```diff
@@ -3,3 +3,4 @@ rules:
   - selectorRegexp: k8s[.]io/kubernetes
     allowedPrefixes:
       - k8s.io/kubernetes/pkg/kubelet/cri
+      - k8s.io/kubernetes/pkg/probe/exec
```

Comment:
Don't have context on why we're restricting internal imports here, can't think of a better location to put TimeoutError though

Reply:
@dims in https://github.com/kubernetes/kubernetes/pull/90552 it seems like this import restrictions file was moved here from another kubelet package. The comment above "prevent exposing internal api in streaming packages" makes me think that this was initially intended for `pkg/kubelet/cri/streaming` but I'm not sure anymore. Could use your guidance on what's best here -- what's the reason for not allowing internal imports to `pkg/kubelet/cri`, were there plans to move that to staging at some point?

Reply:
@andrewsykim yes, we needed `streaming` hermetically sealed :)

Reply:
Okay, so can I move this import restriction file to `pkg/kubelet/cri/streaming` instead of in `pkg/kubelet/cri` where it currently lives? That would allow for some imports I'm trying to add here (`pkg/probe/exec`)

Reply:
👍